### PR TITLE
added example docs

### PR DESCRIPTION
**Pull request by ysqyang**
added example docs

### CODE REVIEW

File: docs/source/examples/cim.rst [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                                                            num_actions=num_actions))
+
+                experience_pool = ColumnBasedStore(**config.agents.experience_pool)
+                agent_dict[agent_id] = CIMAgent(name=agent_id, algorithm=algorithm, experience_pool=experience_pool,
+                                                **config.agents.training_loop_parameters)
+
+Main Loop with Actor and Learner (Single Process)
+-------------------------------------------------
+
+This single-process workflow of a learning policy's interaction with a MARO environment is comprised of:
```
Comment (Jinyu-W) [inline docs/source/examples/cim.rst:165]: tailing with Noun phrase? How about

- Initialization of ...
- Definition of ..
- Creation of ..

File: docs/source/examples/cim.rst [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+.. code-block::python
+
+    env = Env(config.env.scenario, config.env.topology, durations=config.env.durations)
+    agent_id_list = [str(agent_id) for agent_id in env.agent_idx_list]
+    state_shaper = CIMStateShaper(**config.state_shaping)
+    action_shaper = CIMActionShaper(action_space=list(np.linspace(-1.0, 1.0, config.agents.algorithm.num_actions)))
+    if config.experience_shaping.type == "truncated":
+        experience_shaper = TruncatedExperienceShaper(**config.experience_shaping.truncated)
+    else:
+        experience_shaper = KStepExperienceShaper(reward_func=lambda mt: 1-mt["container_shortage"]/mt["order_requirements"],
```
Comment (Jinyu-W) [inline docs/source/examples/cim.rst:183]: space before and after: - /

File: docs/source/examples/cim.rst [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+--------------------------------------------------------------
+
+We demonstrate a single-learner and multi-actor topology where the learner drives the program by telling remote actors
+to perform roll-out tasks and using the results they sent back to improve the policies. The workflow usually involves
+launching a learner process and an actor process separately. Because training occurs on the learner side and inference
+occurs on the actor side, we need to create appropriate agent managers on both sides.
+
+On the actor side, the agent manager must be equipped with all shapers as well as an explorer. Thus, The code for
+creating an environment and an agent manager on the actor side is similar to that for the single-host version,
+except that it is necessary to set the AgentMode to INFERENCE. As in the single-process version, the environment and the
```
Comment (Jinyu-W) [inline docs/source/examples/cim.rst:217]: How about applying the format like ``AgentMode.INFERENCE`` (\`\`AgentMode.INFERENCE\`\`) to key words?

File: docs/source/examples/cim.rst [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+                                    explorer=explorer)
+    proxy_params = {"group_name": config.distributed.group_name,
+                    "expected_peers": config.distributed.actor.peer,
+                    "redis_address": (config.distributed.redis.host_name, config.distributed.redis.port)
+                    }
+    actor_worker = ActorWorker(local_actor=SimpleActor(env=env, inference_agents=agent_manager),
+                               proxy_params=proxy_params)
+    actor_worker.launch()
+
+On the learner side, an agent manager in TRAIN mode is required. However, it is not necessary to create shapers for an
```
Comment (Jinyu-W) [inline docs/source/examples/cim.rst:239]: ``AgentMode.TRAIN`` (\`\`AgentMode.TRAIN\`\`)

File: docs/source/examples/citi_bike.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+bikes. We use a heap data structure to find the top k supply/demand candidates from the action scope associated with
+each decision event.
+
+.. code-block:: python
+
+    class GreedyPolicy:
+        ...
+        def choose_action(self, decision_event: DecisionEvent):
+            if decision_event.type == DecisionType.Supply:
+                # find k target stations with the most empty slots, randomly choose one of them and send as many bikes to
```
Comment (Jinyu-W) [inline docs/source/examples/citi_bike.rst:22]: Start with a capital letter, end with a period.

File: docs/source/examples/citi_bike.rst [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                        continue
+
+                    heapq.heappush(top_k_demands, (available_docks, demand_candidate))
+                    if len(top_k_demands) > self._demand_top_k:
+                        heapq.heappop(top_k_demands)
+
+                max_reposition, target_station_idx = random.choice(top_k_demands)
+                action = Action(decision_event.station_idx, target_station_idx, max_reposition)
+            else:
+                # find k source stations with the most bikes, randomly choose one of them and request as many bikes from
```
Comment (Jinyu-W) [inline docs/source/examples/citi_bike.rst:36]: Start with a capital letter, end with a period.

**Review by Jinyu-W — CHANGES_REQUESTED** [review]

File: docs/source/examples/cim.rst [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
+This example demonstrates how to use MARO's reinforcement learning (RL) toolkit to solve the `container inventory
+management <https://maro.readthedocs.io/en/latest/scenarios/container_inventory_management.html>`_ (CIM) problem. It is
+formalized as a multi-agent reinforcement learning problem, where each port acts as a decision agent. The agents take
+actions independently, e.g., loading containers to vessels or discharging containers from vessels.
+
+State Shaper
+------------
+
+`State shaper <https://maro.readthedocs.io/en/latest/key_components/rl_toolkit.html#shapers>`_ converts the environment
```
[the remark this answers is not in the document]
Comment (ysqyang) [reply docs/source/examples/cim.rst:12]: Sticking to 120 default line width in PyCharm.

File: docs/source/examples/cim.rst (hunk 1 above)
Comment (ysqyang) [reply docs/source/examples/cim.rst:165]: Using initializing, defining, creating ... instead

**Review by Jinyu-W — APPROVED** [review]

**Pull request merged**